android/build.gradle

```diff
@@ -50,8 +50,10 @@ android {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion"
+    implementation "org.jetbrains.kotlin:kotlin-reflect:$kotlinVersion"
```

Comment:
I think it's not very nice to add this dependency. The reflect library is quite big and we should try to avoid it if possible.

Reply:
Yes, it would be nice if the analytics collector was separate, but it's been agreed that its initial release is meant to be part of this wrapper, just like we've done with Ads. And those can be split into different packages in the future.

And regarding the reflection API, I added it in order to simplify mapping the analytics [`customData` fields](https://github.com/bitmovin/bitmovin-analytics-collector-android/blob/main/collector/src/main/java/com/bitmovin/analytics/data/CustomData.kt) from JS <-> Kotlin.

The idea is to loop over all the properties of a JS object to dynamically populate the properties of a Kotlin `CustomData` and vice-versa. So for this, I needed a way to dynamically get/set the properties of a Kotlin object based on its name and the only way I could achieve this was by using reflection. See the [extension file](https://github.com/bitmovin/bitmovin-player-react-native/blob/a14bcca532b5d0c1e62e3b521f799ba799d72189/android/src/main/java/com/bitmovin/player/reactnative/extensions/Any.kt).

Do you think there's any way we could replicate the same behavior without `kotlin-reflect`? If so I would gladly switch the implementation.

Reply:
I also found [this alternative](https://github.com/Kotlin/kotlinx.reflect.lite) to `kotlin-reflect` that states to be lighter so maybe it could be used instead. What do you think?

Reply:
I could achieve the same behavior using just java reflection: 7e0a03865a0714c85d1a0c03e6efe9d9bf47b970. What do you think about this solution? Would it have any potential problems?

My local tests worked just fine, though.